### PR TITLE
Added recipe for git-add and git-reset (staging)

### PR DESCRIPTION
**Pull request by FabriceSalvaire**

### CODE REVIEW

**Comment by FabriceSalvaire** [comment]
Must be fixed to `pygit2.IndexEntry(path, ...`

File: docs/recipes/git-add-reset.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+.. code-block:: python
+
+    >>> index = repo.index
+    >>> index.remove(path)
+    >>> # Skip the following lines if the file is new, go to "Write index"
+    >>> ## Get the entry for this file in HEAD
+    >>> tree_entry = repo.revparse_single('HEAD').tree[path]
+    >>> ## Restore the entry in the index
+    >>> index_entry = pygit2.IndexEntry(tree_entry.name, tree_entry.oid, tree_entry.filemode)
```
Comment (FabriceSalvaire) [inline docs/recipes/git-add-reset.rst]: Fix: tree_entry.name -> path

**Comment by FabriceSalvaire** [comment]
I am wrong, we have to use **Status** http://www.pygit2.org/working-copy.html#status but doc is minimalist

**Comment by jdavid** [comment]
Yes use ``repo.status_file(path)``, which is just a wrapper around ``git_status_file``, see https://libgit2.github.com/libgit2/#HEAD/group/status/git_status_file and https://libgit2.github.com/libgit2/#HEAD/type/git_status_t

The documentation is lacking, any help to improve it is appreciated, thanks!

**Comment by FabriceSalvaire** [comment]
I will remake a PR

**Pull request merged**